liferay-workspace/tests/blade.samples.test/src/testFunctional/java/com/liferay/blade/samples/controlmenu/entry/test/BladeControlMenuEntryTest.java

```diff
@@ -77,61 +62,27 @@ public void testBladeControlMenuEntry()
 
 		Assert.assertTrue(
 			"Control Menu Entry Link is not visible",
-			isVisible(_controlMenuLink));
+			_controlMenuLink.isDisplayed());
 
 		Assert.assertTrue(
-			"Control Menu Entry Link text is not visible",
-			isTextPresent(
-				_controlMenuLinkText, "Blade Menu Entry Custom Message"));
+			"Expected: Blade Menu Entry Custom Message, but saw" +
+				_controlMenuLinkText.getText(),
+			_controlMenuLinkText.getText().contains(
+				"Blade Menu Entry Custom Message"));
+
+		BladeSampleFunctionalActionUtil.mouseOverClick(
+			_webDriver, _controlMenuLinkText);
 
-		customClick(_webDriver, _controlMenuLink);
+		Thread.sleep(1000);
```

Comment:
Does webdriver not have a built-in ability to wait for components to become visible so we don't have to use this?

Reply:
I get runtime exceptions using `webDriverWait.until(ExpectedConditions.visibilityOf(webelement)`

Reply:
so i just removed them. in most cases, the chromeheadless didn't seem to need it anymore. there were just a few cases where the DOM took a bit of waiting to load. In those cases, I dont know if the until() would've matter because it was a new element that appeared.